synapse/http/client.py

```diff
@@ -356,44 +345,11 @@ def __init__(
         # reactor.
         self._cooperator = Cooperator(scheduler=_make_scheduler(hs.get_reactor()))
 
-        if self._ip_blacklist:
-            # If we have an IP blacklist, we need to use a DNS resolver which
-            # filters out blacklisted IP addresses, to prevent DNS rebinding.
-            self.reactor: ISynapseReactor = BlacklistingReactorWrapper(
-                hs.get_reactor(), self._ip_whitelist, self._ip_blacklist
-            )
-        else:
-            self.reactor = hs.get_reactor()
-
-        # the pusher makes lots of concurrent SSL connections to sygnal, and
-        # tends to do so in batches, so we need to allow the pool to keep
-        # lots of idle connections around.
-        pool = HTTPConnectionPool(self.reactor)
-        # XXX: The justification for using the cache factor here is that larger instances
-        # will need both more cache and more connections.
-        # Still, this should probably be a separate dial
-        pool.maxPersistentPerHost = max(int(100 * hs.config.caches.global_factor), 5)
-        pool.cachedConnectionTimeout = 2 * 60
-
-        self.agent: IAgent = ProxyAgent(
+        self.agent: IAgent = Agent(
```

Comment:
We never use this agent, maybe we could add a class-property instead which is just:

```python
class BaseHttpClient:
    agent: IAgent
```

And allow each implementation to create one? (Should probably be in the docstring that sub-classes need to do that.)

Reply:
Do you suppose should add a bit of code to raise an Exception of some kind? I think it will end up in `request` if so.

OTOH, provides a great example of a standard `Agent` with basic browser like behavior.

Reply:
> Do you suppose should add a bit of code to raise an Exception of some kind?

Do you mean if an implementation doesn't define `agent`? I *think* that the static analyzer (`mypy`) will complain in that situation, but not 100% sure.

> OTOH, provides a great example of a standard `Agent` with basic browser like behavior.

Yes -- would any callers use that or would they all redefine it though?

Reply:
No, I don't imagine they would use it.

Reply:
Removed in [445b6e3](https://github.com/matrix-org/synapse/pull/15427/commits/445b6e3421a9b6b9971a86b2d2f9e9db2f4741de)